webextension/js/popup.js

```diff
@@ -1,150 +1,335 @@
+// ----------------------------------------------------------------------------
 
-  // Let"s retrieve the list of tabs.
-  browser.runtime.sendMessage({
-    method: "getTabs",
-    userContextId: identity.userContextId,
-  }).then(tabs => {
-    console.log('browser.runtime.sendMessage getTabs, tabs: ', tabs);
-    // For each one, let's create a new line.
+Logic.registerPanel(P_CONTAINERS_LIST, {
+  panelSelector: "#container-panel",
+
+  // This method is called when the object is registered.
+  initialize() {
+    document.querySelector("#container-add-link").addEventListener("click", () => {
+      Logic.showPanel(P_CONTAINER_EDIT, {});
+    });
+
+    document.querySelector("#edit-containers-link").addEventListener("click", () => {
+      Logic.showPanel(P_CONTAINERS_EDIT);
+    });
+
+    document.querySelector("#sort-containers-link").addEventListener("click", () => {
+      browser.runtime.sendMessage({
+        method: "sortTabs"
+      }).then(() => {
+        window.close();
+      });
+    });
+  },
+
+  // This method is called when the panel is shown.
+  prepare() {
     let fragment = document.createDocumentFragment();
-    for (const tab of tabs) {
+
+    Logic.identities().forEach(identity => {
+      log('identities.forEach');
       let tr = document.createElement("tr");
       fragment.appendChild(tr);
-      tr.classList.add("container-info-tab");
+      tr.classList.add("container-panel-row");
       tr.classList.add("clickable");
       tr.innerHTML = `
-        <td><img class="icon" src="${tab.favicon}" /></td>
-        <td>${tab.title}</td>`;
-      // On click, we activate this tab.
-      tr.addEventListener("click", () => {
-        browser.runtime.sendMessage({
-          method: "showTab",
-          tabId: tab.id,
-        }).then(() => {
-          window.close();
-        });
+        <td>
+          <div class="userContext-icon open-newtab"
+            data-identity-icon="${identity.image}"
+            data-identity-color="${identity.color}">
+          </div>
+        </td>
+        <td class="open-newtab">${identity.name}</td>
+        <td class="info">&gt;</td>`;
+
+      tr.addEventListener("click", e => {
+        if (e.target.matches(".open-newtab")) {
+          browser.runtime.sendMessage({
+            method: "showTabs",
+            userContextId: identity.userContextId
+          }).then(() => {
+            return browser.runtime.sendMessage({
+              method: "openTab",
+              userContextId: identity.userContextId,
+            });
+          }).then(() => {
+            window.close();
+          });
+        } else if (e.target.matches(".info")) {
+          Logic.showPanel(P_CONTAINER_INFO, identity);
+        }
       });
-    }
+    });
 
-    document.getElementById("container-info-table").appendChild(fragment);
-  })
+    document.querySelector(".identities-list").innerHTML = "";
+    document.querySelector(".identities-list").appendChild(fragment);
```

Comment:
Should we cache `document.querySelector(".identities-list")` so we don't need to do a DOM lookup twice?